tools/reviewer/webapp/src/app/dashboard/file-changes/file-changes.component.ts

```diff
@@ -1,40 +1,74 @@
 import { Component, OnDestroy, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 
-import { FileChangesService } from './file-changes.service';
+import { CommitSelectService } from './commit-select';
+import {
+  CommitService,
+  ExtensionService,
+  LoadService,
+  StateService,
+  ThreadService,
+} from './services';
 
-// The component implements file changes page
+// The component implements file-changes page.
+// Frame for code-changes.
 @Component({
   selector: 'file-changes',
   templateUrl: './file-changes.component.html',
   styleUrls: ['./file-changes.component.scss'],
-  providers: [FileChangesService],
+  providers: [
+    LoadService,
+    ExtensionService,
+    ThreadService,
+    CommitService,
+    CommitSelectService,
+  ],
 })
 export class FileChangesComponent implements OnInit, OnDestroy {
-  language: string;
-
   constructor(
     private activatedRoute: ActivatedRoute,
-    private fileChangesService: FileChangesService,
-  ) { }
+    public stateService: StateService,
+    private loadService: LoadService,
+    private extensionService: ExtensionService,
+    public commitService: CommitService,
+    public commitSelectService: CommitSelectService,
+  ) {
+    this.stateService.isLoading = true;
+    this.stateService.isCommitFound = true;
+  }
 
   ngOnInit() {
     this.getUrlParam();
   }
 
   // Get parameters from url
   getUrlParam(): void {
+    this.stateService.diffId = this.activatedRoute.snapshot.url[0].path;
     const filename: string = this.activatedRoute.snapshot.url
       .splice(1)
       .map(v => v.path)
       .join('/');
-    const diffId = this.activatedRoute.snapshot.url[0].path;
+    this.stateService.file.setFilenameWithRepo(filename);
+
+    this.stateService.language = this.extensionService.getLanguage(filename);
+
+    // Get left and right commit ids from url if they present
+    this.activatedRoute.queryParams.subscribe(params => {
+      if (params.left_commit_id) {
```

Comment:
Nice! I didn't know about it

Reply:
> can we use a shorter 10 digit one?
We can, but that means we'll save `10 digit commits` in firebase. So, if we'll decide to use more long version in the future it will be.. difficult. To do this we'll need to bulk update each commit of each thread or even delete the threads.
Another way might be to ask local server to give long version by short version (is that possible?) and then save in firebase long one.
WDYT?